### PR TITLE
Log outgoing botnet traffic and some more incoming

### LINKED ISSUE #22: Console mode +t does not show outbound botnet traffic.
| Trac | Data |
| --: | :-- |
| Ticket | 22 |
| Reported by | Regan Wallace |
| Status | new |
| Component | Core |
| Priority | minor |
| Milestone | 1.8.0 |
| Keywords | log loglevel |
| Version | 1.8.0 CVS |

If on the console, you set .console +t it will only show inbound botnet
traffic.

.console +t should show outbound botnet traffic as well; either that, or have
another mode that shows outbound botnet traffic.

### PR DESCRIPTION
**Pull request by Cizzle**
Found by: Regan Wallace
Patch by: Cizzle
Fixes: #22 

One-line summary: Now logs all raw outgoing botnet traffic and raw outgoing share traffic and also some forgotten incoming botnet traffic.

Additional description: This also fixes a long existin bug where console mode h (raw (incoming) share traffic) didn't actually log anything, but appeared under console mode t (raw (incoming) botnet traffic).

### CODE REVIEW

**Comment by vanosg** [comment]
LGTM :+1:

File: src/flags.c [hunk 1]
```diff
@@ -215,8 +221,10 @@ char *maskname(int x)
     i += my_strcpy(s + i, "wallops, ");
   if ((x & LOG_SRVOUT) && raw_log)
     i += my_strcpy(s + i, "server output, ");
-  if ((x & LOG_BOTNET) && raw_log)
-    i += my_strcpy(s + i, "botnet traffic, ");
+  if ((x & LOG_BOTNETIN) && raw_log)
+    i += my_strcpy(s + i, "botnet traffic in, ");
```
Comment (thommey) [inline src/flags.c]: how about "botnet input"/"botnet output"? or "botnet incoming"/"botnet outgoing"? "botnet traffic in" sounds a bit odd IMHO

Comment (vanosg) [reply src/flags.c]: Concur... botnet incoming/outgoing sounds best to me.

**Comment by vanosg** [comment]
@Cizzle howdy! Do you have a local copy of the description for the raw botnet flags document you drafted, by any chance? Come say hi, we miss you :)

**Pull request merged**